src/hooks/useAutoRAGPolling.ts

```diff
@@ -5,26 +5,25 @@ import {
   AuthService,
 } from "../services/auth-service";
 
-export interface AutoRAGStatus {
-  status: "processing" | "ready" | "error" | "not_found";
-  message: string;
-  tenant?: string;
-  doc?: string;
-  type?: "single_file" | "split_file" | "staging";
-  lastUpdate?: string;
-  timestamp?: string;
+export interface AutoRAGJobStatus {
+  id: string;
+  source: string;
+  end_reason?: string;
+  ended_at?: string;
+  last_seen_at?: string;
+  started_at: string;
 }
 
 export interface UseAutoRAGPollingReturn {
-  status: AutoRAGStatus | null;
+  jobStatus: AutoRAGJobStatus | null;
   isPolling: boolean;
-  startPolling: (tenant: string, filename: string) => void;
+  startPolling: (ragId: string, jobId: string) => void;
   stopPolling: () => void;
   error: string | null;
 }
 
 export function useAutoRAGPolling(): UseAutoRAGPollingReturn {
-  const [status, setStatus] = useState<AutoRAGStatus | null>(null);
+  const [jobStatus, setJobStatus] = useState<AutoRAGJobStatus | null>(null);
```

Comment:
just a reminder to have safety checks in places + account for this null. hopefully you are :)